### PR TITLE
chore: refactored code to use custom hooks

### PR DESCRIPTION
**Pull request by azeezat**
# Description

Extracted most of the functions in the index.ts file into custom react hooks to improve readability. Feel free to check and comment on this PR. Everything should work as before so if you notice any changes, please drop a comment.

- [x] Code cleanup

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
Added unit tests to ensure that the library continues to function as it currently does.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/hooks/use-modal.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  useEffect(() => {
+    if (hideModal) {
+      setOpen(false);
+    }
+  }, [hideModal]);
+
+  useEffect(() => {
+    if (!open && Platform.OS === 'android') {
+      onDismiss?.();
```
[the remark this answers is not in the document]
Comment (azeezat) [reply src/hooks/use-modal.ts]: @kofiarkoh Thank you for pointing this out. I just added some tests to address this.

[the remark this answers is not in the document]
Comment (azeezat) [reply src/hooks/use-modal.ts]: @kofiarkoh How do you currently use the `hideModal` prop? I did some refactoring on the modal and I haven't been able to find a use case for it

**Review by Tabspear — COMMENTED** [review]
Clean Code!

**Review by Tabspear — APPROVED** [review]
Great Job!

**Pull request merged**